### PR TITLE
Support Subject Name/Issuer authentication

### LINKED ISSUE #10816: Support SubjectName/Issuer (SNI) authentication
See AzureAD/microsoft-authentication-library-for-python#60 for a description and AzureAD/microsoft-authentication-library-for-python#71 for MSAL's implementation.

### PR DESCRIPTION
**Pull request by chlowell**
This leverages MSAL to support SNI auth with the sync `CertificateCredential` (closes #10816). I've tested this manually and will add a live test in a later PR. Adding equivalent support to the async `CertificateCredential` is tracked by #13349.

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_credentials/certificate.py [hunk 1]
```diff
@@ -54,9 +56,30 @@ def __init__(self, tenant_id, client_id, certificate_path, **kwargs):
 
         # TODO: msal doesn't formally support passwords (but soon will); the below depends on an implementation detail
         private_key = serialization.load_pem_private_key(pem_bytes, password=password, backend=default_backend())
+        client_credential = {"private_key": private_key, "thumbprint": hexlify(fingerprint).decode("utf-8")}
+        if kwargs.pop("send_certificate", False):
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_credentials/certificate.py:60]: Should we do this only in
if kwargs.pop("send_certificate", False):
else:
    private_key = serialization.load_pem_private_key(pem_bytes, password=password, backend=default_backend())
    client_credential = {"private_key": private_key, "thumbprint": hexlify(fingerprint).decode("utf-8")}

File: sdk/identity/azure-identity/azure/identity/_credentials/certificate.py [hunk 2]
```diff
@@ -54,9 +56,30 @@ def __init__(self, tenant_id, client_id, certificate_path, **kwargs):
...
         super(CertificateCredential, self).__init__(
-            client_id=client_id,
-            client_credential={"private_key": private_key, "thumbprint": hexlify(fingerprint).decode("utf-8")},
-            tenant_id=tenant_id,
-            **kwargs
+            client_id=client_id, client_credential=client_credential, tenant_id=tenant_id, **kwargs
         )
+
+
+def extract_cert_chain(pem_bytes):
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_credentials/certificate.py:75]: public method?

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/certificate.py:75]: Private package.

File: sdk/identity/azure-identity/azure/identity/_credentials/certificate.py (hunk 1 above)
Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/certificate.py:60]: We need the private key and fingerprint in both cases. `send_certificate=True` means we need to include the cert content as well.

File: sdk/identity/azure-identity/azure/identity/_credentials/certificate.py (hunk 2 above)
Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_credentials/certificate.py:75]: Yes. But if we only want to use the method in this model, shouldn't we use _extract_cert_chain? Or we may want to use it in other models in the future?

**Pull request merged**